### PR TITLE
Adding error handling for iOS

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
- [x] Adds slack error handling for iOS.
- [x] Migrates logic for checking for outdated versions to the backend.
- [x] Removes unused constants.
- [x] Simplifies the codebase.

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

File: mobile/lib/services/rest_api.dart [hunk 1]
```diff
@@ -114,6 +111,33 @@ class AirqoApiClient {
...
+        headers: {
+          HttpHeaders.contentTypeHeader: 'application/json',
+        },
+        body: jsonEncode({
+          'text': "Exception: ${exception.toString()}\n\n "
+              "App details: $packageInfo\n\n "
+              "StackTrace: $stackTrace\n\n",
+        }),
+      );
+    } catch (e) {
```
Comment (NoahNsimbe) [inline mobile/lib/services/rest_api.dart:136]: Sending errors slack

File: mobile/lib/utils/exception.dart [hunk 2]
```diff
@@ -24,9 +27,20 @@ Future<void> logException(
...
   try {
-    FirebaseCrashlytics.instance
-        .recordError(exception, stackTrace, fatal: true);
+    if (!Platform.isAndroid) {
+      if (fatal) {
+        await AirqoApiClient.sendErrorToSlack(exception as Object, stackTrace);
+      }
+
+      return;
+    }
```
Comment (NoahNsimbe) [inline mobile/lib/utils/exception.dart:36]: Sending iOS errors to slack

File: mobile/lib/constants/config.dart [hunk 3]
```diff
@@ -59,12 +64,6 @@ class Config {
   static String get airqoSecondaryLogo =>
       'https://storage.cloud.google.com/airqo-app/public-images/airqo_logo.png';
 
-  static String get placesSearchUrl =>
-      'https://maps.googleapis.com/maps/api/place/';
-
-  static String get appStoreUrl =>
-      'https://apps.apple.com/ug/app/airqo-monitoring-air-quality/id1337573091';
```
Comment (NoahNsimbe) [inline mobile/lib/constants/config.dart:66]: cleanup

File: mobile/lib/services/app_service.dart [hunk 4]
```diff
@@ -109,35 +106,4 @@ class AppService {
       ),
     );
   }
-
-  Future<AppStoreVersion?> latestVersion() async {
-    AppStoreVersion? appStoreVersion;
-
-    try {
-      final PackageInfo packageInfo = await PackageInfo.fromPlatform();
-
```
Comment (NoahNsimbe) [inline mobile/lib/services/app_service.dart:118]: logic transferred to API

File: mobile/lib/services/rest_api.dart [hunk 5]
```diff
@@ -1,29 +1,25 @@
...
-      (key, value) {
-        url = queryParams.keys.first.compareTo(key) == 0
-            ? '$url$key=$value'
-            : '$url&$key=$value';
-      },
-    );
-  }
+  String formattedUrl = '$url?TOKEN=${Config.airqoApiV2Token}';
+  queryParams
+      .forEach((key, value) => formattedUrl = "$formattedUrl&$key=$value");
```
Comment (NoahNsimbe) [inline mobile/lib/services/rest_api.dart:20]: Simplifying the code

File: mobile/lib/utils/extensions.dart [hunk 6]
```diff
@@ -673,41 +672,6 @@ extension FileExt on File {
...
-
-    // checking first code
-    if (versionSections.first > candidateSections.first) return 1;
-
-    if (versionSections.first < candidateSections.first) return -1;
-
-    // checking second code
-    if (versionSections[1] > candidateSections[1]) return 1;
-
-    if (versionSections[1] < candidateSections[1]) return -1;
```
Comment (NoahNsimbe) [inline mobile/lib/utils/extensions.dart:700]: Logic transferred to the APIs

**Review by Baalmart — APPROVED** [review]
thanks @NoahNsimbe , this looks good to me! Thanks!
Based on the details of the PR, I will just go ahead and merge it to make things timely!

Thanks!

**Pull request merged**